crates/ra_hir_ty/Cargo.toml

```diff
@@ -27,9 +27,14 @@ test_utils = { path = "../test_utils" }
 
 scoped-tls = "1"
 
-chalk-solve = { version = "0.15.0" }
-chalk-ir = { version = "0.15.0" }
+chalk-solve = { version = "0.17.0-dev.0", git = "https://github.com/rust-lang/chalk.git" }
+chalk-ir = { version = "0.17.0-dev.0", git = "https://github.com/rust-lang/chalk.git" }
+chalk-recursive = { version = "0.17.0-dev.0", git = "https://github.com/rust-lang/chalk.git" }
 
 [dev-dependencies]
 insta = "0.16.0"
 expect = { path = "../expect" }
+
+tracing = "0.1"
+tracing-subscriber = { version = "0.2" }
```

Comment:
You might be able to disable _all_ default features and only enable the EnvFilter + the Registry to cut down on compile times, as you don't really need the formatters.

Reply:
I wonder if we can import only a single trait, like how we need only env_logger for logging. The problem with many crates is that you need to update several major versions sometimes.

Reply:
I've been thinking about that—one thing that I can do is re-export the important stuff from `tracing-subscriber` via `tracing-tree`, which could reduce the amount of imports to just tracing and tracing-tree.

Reply:
I set the features, but I'm not sure it made any difference :thinking: